beacon_node/network/src/sync/block_lookups/common.rs

```diff
@@ -53,115 +43,77 @@ pub trait RequestState<T: BeaconChainTypes> {
     /// The type created after validation.
     type VerifiedResponseType: Clone;
 
-    /// We convert a `VerifiedResponseType` to this type prior to sending it to the beacon processor.
-    type ReconstructedResponseType;
-
-    /* Request building methods */
-
-    /// Construct a new request.
-    fn build_request(
-        &mut self,
-        lookup_type: LookupType,
-    ) -> Result<(PeerId, Self::RequestType), LookupRequestError> {
-        // Verify and construct request.
-        self.too_many_attempts(lookup_type)?;
-        let peer = self.get_peer()?;
-        let request = self.new_request();
-        Ok((peer, request))
-    }
-
-    /// Construct a new request and send it.
-    fn build_request_and_send(
+    /// Potentially makes progress on this request if it's in a progress-able state
+    fn continue_request(
         &mut self,
         id: Id,
-        lookup_type: LookupType,
+        awaiting_parent: bool,
```

Comment:
I updated the bool args to enums in my PR just so you can't mix up the bools as you pass them into the function https://github.com/dapplion/lighthouse/pull/26/